src/score_hv/harvesters/soca_diags.py

```diff
@@ -0,0 +1,301 @@
+
+            print("file is open ",filename)
+            """
+              The information about the file from the 
+              file name.
+              """
+            variable = filename_info['variable_type']  
+            filetime_str = filename_info['datetime']
+            dt_obj = dt.strptime(filetime_str, "%Y%m%d%H")
+            filetime = dt_obj.strftime("%Y-%m-%d %H:%M:%S")
+            sensor = filename_info['sensor']
+            file_region = filename_info['region']
+            satellite = filename_info['satellite']
+            level = filename_info['level']
+            latitude,longitude = read_MetaData_group(dataset)
+            
+            """
+              We can now loop through the list of wanted groups.
+              We get the values from the wanted groups and the FillValue.
+              """
+            for group in groups_wanted:
+                if group not in dataset.groups:
+                   print(group," is not in the dataset.groups ",dataset.groups)
+                   sys.exit(1)
+                else:     
+                   requested_group = dataset.groups[group]
+                   group_name = group
+                   num_variables = len(requested_group.variables)
+                   for var_name in requested_group.variables:
+                       variable = var_name
+                       var = requested_group.variables[var_name]
+                       if num_variables > 1:
+                          longname = var_name 
+                       else:
+                          longname = list(requested_group.variables.keys())[0] 
+
+                       if "units" in var.ncattrs():
+                          units = str(var.getncattr("units"))
+   
+                       groupvalue_variable = requested_group.variables[var_name]
+                       var_values = np.array(groupvalue_variable[:])
+
+                       if '_FillValue' in groupvalue_variable.ncattrs():
+                          fill_value = groupvalue_variable.getncattr('_FillValue')
+
+                       if np.ma.isMaskedArray(var_values):
+                          var_values = np.ma.filled(var_values, np.nan) 
+                       else:   
+                          var_values[var_values == fill_value] = np.nan                                       
+                       
+                       """
+                          Calculate the requested statistics. Our var_values arrays are 
+                          all of type <class 'numpy.ndarray'>.  We have used the fill_value
+                          to put np.nan's in the for all values to to the fill_values so we 
+                          can calculate the statistics.
+                          """
+                       for j, statistic in enumerate(self.config.get_stats()):
+                           group = group_name
+                           if statistic == 'mean':
+                               value = np.nanmean(var_values)
```

Comment:
For calculating statistics on arrays that have missing data, I suggest we homogenize var_values so that they are masked arrays with fill values, then use the numpy.ma methods (ma.mean(), ma.median(), etc...) for calculations. To do this safely, we could convert input data (var_values) to a numpy masked array using the numpy.ma.filled method for any given input data, which would retain fill values but apply a mask to the nans.